### ISSUE TITLE
Any timeline on training code?

### ISSUE THREAD

**Issue by liydaco** [SesameAILabs/csm#41]
Congrats on the release, this is the most exciting tech in AI at the moment.

Can you give us any information on the training code / model weights?

**Comment by liydaco** [comment]
> the commit was just reversed - no weights have been dropped yet either

Weird, the hugging face links were broken anyway - I suspect they are updating hf and will push again soon.

**Comment by davidkariuki123** [comment]
What was committed (before it was reversed)?

**Comment by MillionthOdin16** [comment]
You scared em

**Comment by OneGiantChicken** [comment]
> What was committed (before it was reversed)?

If you look in the About section > Forks, there are several copies of what was committed.

**Comment by liydaco** [comment]
> What was committed (before it was reversed)?

https://github.com/liydaco/sesame-csm

**Comment by davidkariuki123** [comment]
> > What was committed (before it was reversed)?
> 
> https://github.com/liydaco/sesame-csm

Thanks! Damn, so it seems they are only releasing the 1B model and not the 8B. I would imagine Maya and Miles is built on top of the 8B model.

**Comment by davidkariuki123** [comment]
> maya and miles was never beeing released they already said that - i think they dont have permission or dont want to opensource that as they acquired that for there product ( they seem to be a hardware company)

Yes I know Maya and Miles weren't being released. I just hoped they would release the base model for Maya and Miles which is presumably the 8B model, not the 1B.

**Comment by davidkariuki123** [comment]
Yes, although it is still the same architecture used in Maya and Miles. It's still their CSM, just the smallest variant.

**Comment by Tiger14n** [comment]
I'm so confused of how this works, i thought it was single model the processes information and generates speech, but it seems the model only generates speech and we have to rely on a separate model/LLM to process and generate the text. still nice but certainly not for running locally.

**Comment by digital-soul18** [comment]
its actually ideal if its only generating speech, allows us to be flexible in using the right type of LLMs that you want. They already have a whispter plus SST in their other repo which I guess is what they use for the speech recognition

**Comment by Lex-au** [comment]
> I'm so confused of how this works, i thought it was single model the processes information and generates speech, but it seems the model only generates speech and we still have to rely on a separate model/LLM to process and generate the text, and another model for speech recognition. still nice but certainly not for running locally.

They dropped the checkpoint on HF. It's a pickled tensor, so we can deserialise it to inspect the full architecture.

**Comment by sambalshikhar** [comment]
CSM seems very slow on A6000 GPU , around ~10s for a single sentence .
Checkout out [LLMVoX](https://mbzuai-oryx.github.io/LLMVoX/)  .
An LLM agnostic ultra fast streaming TTS , language adaptable .

**Comment by juangon** [comment]
@darkacorn do you know if there is any code to finetune kokoro?

**Comment by sambalshikhar** [comment]
@darkacorn We are working in the audio quality , however did you find latency as a problem ?

**Comment by sambalshikhar** [comment]
@darkacorn If you check our paper , mini-cpm has far higher word error rate and latency , Moshi also has very high word error rate than us . We working on improving the audio quality and creating a on device version soon . Thank you for the feedback though .

**Comment by ichDaheim** [comment]
> > [@darkacorn](https://github.com/darkacorn) do you know if there is any code to finetune kokoro?
> 
> kokoro is styletts - so you can use the training code there

@darkacorn: have you tried this? I don't think it will work because hexgrad used istftnet for kokore unlike styletts - at least that's what i grabbed from a discussion on HF before the got closed. If you found a why to fintune/train something alike Kokore, i would be highly interrested.

**Comment by jjyuhub** [comment]
> CSM seems very slow on A6000 GPU , around ~10s for a single sentence . Checkout out [LLMVoX](https://mbzuai-oryx.github.io/LLMVoX/) . An LLM agnostic ultra fast streaming TTS , language adaptable .

LLMVoX is absolutely awful when it comes to passing as a human speaker! It sounds so robotic that it's immediately obvious that it's AI-generated. The intonation is flat, the pauses feel unnatural, and the pronunciation often lacks the fluidity of real human speech.

**Comment by BornSaint** [comment]
> also speed isnt the issue here csm can run 3.4x realtime on a single 3090 .. if you optimize - probably more if someone takes mroe time then me on the inf

why you don't simply explain how did you optimized the code instead of tell this in every issue?

**Comment by MillionthOdin16** [comment]
https://github.com/canopyai/Orpheus-TTS

What CSM release and support and examples should have been

**Comment by TheAjaykrishnanR** [comment]
> https://github.com/canopyai/Orpheus-TTS
> 
> What CSM release and support and examples should have been

thanks for sharing this dude, this looks so fricking cool !

**Comment by jjyuhub** [comment]
> > https://github.com/canopyai/Orpheus-TTS
> > What CSM release and support and examples should have been
> 
> thanks for sharing this dude, this looks so fricking cool !

I'm not sure if it sounds more human?! It doesn't seem to vary the tones and inflections a lot, or does it?

**Comment by TheAjaykrishnanR** [comment]
> > > https://github.com/canopyai/Orpheus-TTS
> > > What CSM release and support and examples should have been
> > 
> > 
> > thanks for sharing this dude, this looks so fricking cool !
> 
> I'm not sure if it sounds more human?! It doesn't seem to vary the tones and inflections a lot, or does it?

there's a 4 bit quant and it sounds like this : https://vocaroo.com/1nskql6J7G5F, this is incredible

**Comment by Tiger14n** [comment]
> https://github.com/canopyai/Orpheus-TTS
> 
> What CSM release and support and examples should have been

Well, i could say for sure we now have an open source Elevenlabs level TTS. But CSM still has an advantage for conversational AI modeling, which is using audio for context

**Comment by Tiger14n** [comment]
> https://github.com/canopyai/Orpheus-TTS
> 
> What CSM release and support and examples should have been

Well, I could now say sure that we have an open source Elevenlabs level TTS.

What I'm excited for is this part, "We use the exact same architecture, and training method, to train end-to-end speech models and we'll probably release an open source end-to-end speech model in the coming weeks."

**Comment by jjyuhub** [comment]
> https://vocaroo.com/1nskql6J7G5F

That's a nice bedtime story tone, but I meant, if it can scream, cry, laugh, rage and so on?

**Comment by TheAjaykrishnanR** [comment]
> > https://vocaroo.com/1nskql6J7G5F
> 
> That's a nice bedtime story tone, but I meant, if it can scream, cry, laugh, rage and so on?

it does if you search the reddit thread, i can post it if you want but NSFW warning: https://vocaroo.com/1otgn5bLIu27

remember this is a 4 bit quant that is reduced to a 2.3 gig gguf taking about 4 gigs in vram

**Comment by AyushBarik** [comment]
if someone makes a finetuning script for this 🙏 🙏

**Comment by 1mono2** [comment]
I want to reproduce the training code for this Sesame model. I'm not very familiar with machine learning, so what should I do first? Can you give me some hints?

I read this article.
https://www.sesame.com/research/crossing_the_uncanny_valley_of_voice

**Comment by darkacorn** [comment]
training script wont be provided they said them self - there are some efforts that may result into a training script .. but nothing concrete rn
